apps/internal/oauth/ops/accesstokens/tokens.go

```diff
@@ -183,6 +183,64 @@ type TokenResponse struct {
 	scopesComputed bool
 }
 
+func (tr *TokenResponse) UnmarshalJSON(data []byte) error {
+	type Alias TokenResponse
+	aux := &struct {
+		ExpiresIn json.Number `json:"expires_in"`
+		ExpiresOn json.Number `json:"expires_on"`
```

Comment:
You could simplify this method by continuing to use `internalTime.DurationTime` to calculate ExpiresOn from `expires_in`:
```suggestion
		ExpiresOn           json.Number               `json:"expires_on,omitempty"`
		ExpiresOnCalculated internalTime.DurationTime `json:"expires_in,omitempty"`
```
then after unmarshaling, if `aux.ExpiresOn` is nonzero, you can convert it to a `time.Time` for the `TokenResponse`. If that field is zero, `ExpiresOnCalculated.T` is the `time.Time` you want